neurodiffeq/generators.py

```diff
@@ -269,15 +278,29 @@ def __init__(self, grid=(10, 10), xy_min=(0.0, 0.0), xy_max=(1.0, 1.0), method='
             grid_x, grid_y = torch.meshgrid(x, y, indexing='ij')
             self.grid_x, self.grid_y = grid_x.flatten(), grid_y.flatten()
             self.getter = lambda: (self.grid_x, self.grid_y)
+        elif method == 'chebyshev2-noisy':
+            self.getter = self.generate(xy_min, xy_max, grid, method='chebyshev2-noisy')
         elif method == 'latin-hypercube':
-            x = _latin_hypercube(xy_min[0], xy_max[0], grid[0])
-            y = _latin_hypercube(xy_min[1], xy_max[1], grid[1])
+            x = lambda: _latin_hypercube(xy_min[0], xy_max[0], grid[0])
```

Comment:
I don't think we want to assign `x` and `y` as functions. The following line calling `torch.meshgrid(x, y, indexing='ij')` will likely fail, unless there's some hacky trick in pytorch I'm not familiar with.